### PR TITLE
fix(pm,lpm): resolved a couple of bugs, make PM more extensible

### PR DESCRIPTION
**Pull request by kyranet**

### CODE REVIEW

File: packages/discord.js-utilities/src/lib/PaginatedMessage.ts [hunk 1]
```diff
@@ -144,89 +152,126 @@ export class PaginatedMessage {
...
 		clone.response = this.response;
-		clone.collector = this.collector;
 		return clone;
 	}
 
-	protected async handleCollect(author: User, channel: TextChannel | NewsChannel, reaction: MessageReaction, user: User) {
+	/**
+	 * Sets up the message.
+	 * @param channel The channel the handler is running at.
+	 * @param author The author the haandler is for.
```
Comment (vladfrangu) [inline packages/discord.js-utilities/src/lib/PaginatedMessage.ts:227]: q: I assume the user is there for extensions?

**Review by vladfrangu — APPROVED** [review]

File: packages/discord.js-utilities/src/lib/PaginatedMessage.ts (hunk 1 above)
Comment (jurienhamaker) [inline packages/discord.js-utilities/src/lib/PaginatedMessage.ts:227]:
```suggestion
	 * @param author The author the handler is for.
```

File: packages/discord.js-utilities/src/lib/PaginatedMessage.ts [hunk 2]
```diff
@@ -144,89 +152,126 @@ export class PaginatedMessage {
...
+	protected async setUpMessage(channel: TextChannel | NewsChannel): Promise<void> {
+		const firstPage = this.messages[this.index]!;
+		if (this.response) await this.response.edit(firstPage);
+		else this.response = (await channel.send(firstPage)) as Message;
+	}
+
+	/**
+	 * Sets up the message's reactions and the collector.
+	 * @param channel The channel the handler is running at.
+	 * @param author The author the haandler is for.
```
Comment (jurienhamaker) [inline packages/discord.js-utilities/src/lib/PaginatedMessage.ts:239]:
```suggestion
	 * @param author The author the handler is for.
```

File: packages/discord.js-utilities/src/lib/PaginatedMessage.ts (hunk 1 above)
Comment (kyranet) [reply packages/discord.js-utilities/src/lib/PaginatedMessage.ts:227]: Yes, it is.

**Review by Favna — APPROVED** [review]

**Pull request merged**